pyuvdata/uvdata/uvh5.py

```diff
@@ -189,256 +194,619 @@ def _get_compression(compression):
+
+    _int_attrs = frozenset(
+        {
+            "Nblts",
+            "Ntimes",
+            "Npols",
+            "Nspws",
+            "Nfreqs",
+            "uvplane_reference_time",
+            "Nphase",
+            "Nants_data",
+            "Nants_telescope",
+        }
+    )
+    _float_attrs = frozenset(
+        {
+            "dut1",
+            "earth_omega",
+            "gst0",
+            "phase_center_ra" "phase_center_dec",
+            "phase_center_epoch",
+        }
+    )
+    _bool_attrs = frozenset(("flex_spw",))
+
+    def __init__(
+        self,
+        path: str | Path,
+        blt_order: Literal["determine"] | tuple[str] | None = None,
+        blts_are_rectangular: bool | None = None,
+        time_first: bool | None = None,
     ):
+        self.path = Path(path)
+        self.__file = None
+        self.__blts_are_rectangular = blts_are_rectangular
+        self.__time_first = time_first
+        self.__blt_order = blt_order
+
+        self.open()
+
+    def __del__(self):
+        """Close the file when the object is deleted."""
+        if self.__file:
+            self.__file.close()
+
+    def close(self):
+        """Close the file."""
+        self.__header = None
+        if self.__file:
+            self.__file.close()
+        self.__file = None
+
+    def open(self):  # noqa: A003
+        """Open the file."""
+        if self.__file is None:
+            self.__file = h5py.File(self.path, "r")
+            self.__header = self.__file["/Header"]
+
+    @cached_property
+    def header(self) -> h5py.Group:
+        """Get the header group."""
+        if self.__file is None:
+            self.open()
+        return self.__header
+
+    def __getattr__(self, name: str) -> Any:
+        """Get attribute directly from header group."""
+        try:
+            x = self.header[name][()]
+            if name in self._string_attrs:
+                x = bytes(x).decode("utf8")
+            elif name in self._int_attrs:
+                x = int(x)
+            elif name in self._float_attrs:
+                x = float(x)
+
+            self.__dict__[name] = x
+            return x
+        except KeyError:
+            try:
+                return self._defaults[name]
+            except KeyError as e:
+                raise AttributeError(f"{name} not found in {self.path}") from e
+
+    @cached_property
+    def blt_order(self) -> tuple[str]:
+        """Tuple defining order of blts axis."""
+        if self.__blt_order not in (None, "determine"):
+            return self.__blt_order
+
+        h = self.header
+        if "blt_order" in h:
+            blt_order_str = bytes(h["blt_order"][()]).decode("utf8")
+            return tuple(blt_order_str.split(", "))
+        else:
+            if self.__blt_order == "determine":
+                return self.get_blt_order()
+            else:
+                return None
+
+    def get_blt_order(self) -> tuple[str]:
```

Comment:
Can we move this above the `blt_order` method so that it is defined before it is called?

Reply:
Yes, done.